zkevm-circuits/src/evm_circuit/execution/end_tx.rs

```diff
@@ -260,18 +239,27 @@ mod test {
         //     Some(65000),
         //     None,
         // )]);
+
         // Multiple txs
-        test_ok(vec![
-            mock_tx(
-                address!("0x00000000000000000000000000000000000000fd"),
+        test_ok(
+            // Get the execution steps from the external tracer
+            TestContext::<2, 1>::new(
```

Comment:
I don't see the need now considering the cases that I see implemented for the tests.

If at any point it becomes required, anyone that needs it can add it IMO.

Reply:
This was trying to test if `EndTx` transit to `BeginTx` is okay (`tx_id` should be increased by `1`) actually, so perhaps we can still keep it.

Reply:
Should the second tx be dummy completely??

Reply:
I assume dummy tx means minimal tx, actually any valid tx (could generate trace) would work.

Reply:
@han0110 Added the following extra Tx:
```rust
test_ok(
    // Get the execution steps from the external tracer
    TestContext::<2, 2>::new(
        None,
        |accs| {
            accs[0]
                .address(address!("0x00000000000000000000000000000000000000fe"))
                .balance(Word::from(10u64.pow(19)));
            accs[1]
                .address(address!("0x00000000000000000000000000000000000000fd"))
                .balance(Word::from(10u64.pow(19)));
        },
        |mut txs, accs| {
            txs[0]
                .to(accs[0].address)
                .from(accs[1].address)
                .value(Word::from(10u64.pow(17)));
            // Extra TX added here.
            txs[1]
                .to(accs[1].address)
                .from(accs[0].address)
                .value(Word::from(10u64.pow(17)));
        },
        |block, _tx| block.number(0xcafeu64),
    )
    .unwrap()
    .into(),
);
```

This leads to the following circuit verification err:
```
left: `Err([Lookup { name: "LOOKUP: \"RW table\"", lookup_index: 136, location: InRegion { region: Region { index: 5, name: "Execution step" }, offset: 32 } }, Lookup { name: "LOOKUP: \"RW table\"", lookup_index: 136, location: InRegion { region: Region { index: 5, name: "Execution step" }, offset: 48 } }, Lookup { name: "LOOKUP: \"RW table\"", lookup_index: 137, location: InRegion { region: Region { index: 5, name: "Execution step" }, offset: 48 } }, Lookup { name: "LOOKUP: \"RW table\"", lookup_index: 140, location: InRegion { region: Region { index: 5, name: "Execution step" }, offset: 32 } }, Lookup { name: "LOOKUP: \"RW table\"", lookup_index: 141, location: InRegion { region: Region { index: 5, name: "Execution step" }, offset: 32 } }])`,
```

Reply:
As discussed in private, it's solved and the TestContext updated to prevent new cases like this one!